app/src/main/java/lt/vilnius/tvarkau/fragments/NewReportFragment.kt

```diff
@@ -440,38 +439,51 @@ class NewReportFragment : BaseFragment(),
                     val geocoder = Geocoder(context)
                     val place = PlacePicker.getPlace(context, data)
 
-                    var addresses = listOf<Address>()
+                    val addresses = mutableListOf<Address>()
 
                     place.latLng?.let {
                         locationCords = it
 
                         try {
-                            addresses = geocoder.getFromLocation(it.latitude, it.longitude, 1)
-                        } catch (e: IOException) {
-                            Timber.e(e)
+                            geocoder.getFromLocation(it.latitude, it.longitude, 1)?.let {
+                                addresses.addAll(it)
+                            }
+                        } catch (e: Throwable) {
+                            Timber.e(GeocoderException(e))
                         }
                     }
 
                     if (addresses.isNotEmpty()) {
                         if (addresses.first().locality != null) {
```

Comment:
At this point - nothing. How should we approach this? Maybe move this one line up
`addresses.firstOrNull()?.locality != null`?